src/Compilers/VisualBasic/Portable/Language/LanguageFeature.vb

```diff
@@ -0,0 +1,288 @@
+        CObjInAttributeArguments
+        LineContinuationComments
+        TypeOfIsNot
+        YearFirstDateLiterals
+        WarningDirectives
+        PartialModules
+        PartialInterfaces
+        ImplementingReadonlyOrWriteonlyPropertyWithReadwrite
+        DigitSeparators
+        BinaryLiterals
+        Tuples
+        IOperation
+        InferredTupleNames
+    End Enum
+
+    <HideModuleName>
+    Friend Module FeatureExtensions
+
+        <Extension>
+        Friend Function IsPrototype(feature As Feature) As Boolean
+            Select Case feature
+                Case Feature.IOperation
+                    Return True
+            End Select
+            Return False
+        End Function
+
+        <Extension>
+        Friend Function GetFeatureFlag(feature As Feature) As String
+            Select Case feature
+                Case Feature.IOperation
+                    Return "IOperation"
+
+                Case Else
+                    Return Nothing
+            End Select
+        End Function
+
+        <Extension>
+        Friend Function GetLanguageVersion(feature As Feature) As LanguageVersion
+
+            Select Case feature
+                Case Feature.AutoProperties,
+                     Feature.LineContinuation,
+                     Feature.StatementLambdas,
+                     Feature.CoContraVariance,
+                     Feature.CollectionInitializers,
+                     Feature.SubLambdas,
+                     Feature.ArrayLiterals
+                    Return LanguageVersion.VisualBasic10
+
+                Case Feature.AsyncExpressions,
+                     Feature.Iterators,
+                     Feature.GlobalNamespace
+                    Return LanguageVersion.VisualBasic11
+
+                Case Feature.NullPropagatingOperator,
+                     Feature.NameOfExpressions,
+                     Feature.InterpolatedStrings,
+                     Feature.ReadonlyAutoProperties,
+                     Feature.RegionsEverywhere,
+                     Feature.MultilineStringLiterals,
+                     Feature.CObjInAttributeArguments,
+                     Feature.LineContinuationComments,
+                     Feature.TypeOfIsNot,
+                     Feature.YearFirstDateLiterals,
+                     Feature.WarningDirectives,
+                     Feature.PartialModules,
+                     Feature.PartialInterfaces,
+                     Feature.ImplementingReadonlyOrWriteonlyPropertyWithReadwrite
+                    Return LanguageVersion.VisualBasic14
+
+                Case Feature.Tuples,
+                    Feature.BinaryLiterals,
+                    Feature.DigitSeparators
+                    Return LanguageVersion.VisualBasic15
+
+                Case Feature.InferredTupleNames
+                    Return LanguageVersion.VisualBasic15_3
+                Case Else
+                    If feature.IsPrototype Then Return CType(Integer.MinValue, LanguageVersion)
```

Comment:
Is this method ever called with `Feature.IOperation`?

Reply:
It could (prior to recent pushes), via `IsAvailable()`. 
Swapped the order of the checks inside it so the more common, feature released from a specific language version is now first. Hence now the guard for the less common prototype features, which are enabled via feature flags.

Reply:
Initiation point is in `VisualBasicCompilation.vb` line 2724